packages/product-details/src/FreightSimulator.js

```diff
@@ -0,0 +1,66 @@
+import React, { Component } from 'react'
+import { injectIntl, intlShape } from 'react-intl'
+
+import './freight-simulator.css'
+
+/**
+ * Freight simulator component
+ *
+ * Display an input for the zipcode
+ */
+class FreightSimulator extends Component {
+  static propTypes = {
+    intl: intlShape.isRequired,
+  }
+
+  state = {
+    zipcodeValue: '',
+  }
+
+  handleChange = e => {
+    let zipcodeValue = e.target.value || ''
+
+    zipcodeValue = zipcodeValue.replace(/\D/g, '')
+
+    if (zipcodeValue.length > 8) {
+      zipcodeValue = zipcodeValue.substring(0, 8)
+    }
+
+    if (zipcodeValue.length > 5) {
+      zipcodeValue = zipcodeValue.replace(/(\d{5})(\d+)/, '$1-$2')
+    }
+
+    this.setState({
+      zipcodeValue,
+    })
+  }
+
+  handleClick = e => {
+    e.preventDefault()
+
+    // TODO: call graphql
+  }
+
+  formatMessage = id => {
+    return this.props.intl.formatMessage({ id })
+  }
```

Comment:
:pick: No need. Why don't you use the `<FormattedMessage />` ?

Reply:
Because the `<FormattedMessage />` puts an additional `span` tag, and I want the text to be a direct children of `label` for accessibility purpose